src/harness/fakes.ts

```diff
@@ -0,0 +1,376 @@
+                        }
+                    }
+                    catch { /*ignored*/ }
+                }
+            }
+            catch { /*ignored*/ }
+            return { files, directories };
+        }
+
+        public exit(exitCode?: number) {
+            this.exitCode = exitCode;
+            throw processExitSentinel;
+        }
+
+        public getFileSize(path: string) {
+            const stats = this._getStats(path);
+            return stats && stats.isFile() ? stats.size : 0;
+        }
+
+        public resolvePath(path: string) {
+            return vpath.resolve(this.vfs.cwd(), path);
+        }
+
+        public getExecutingFilePath() {
+            if (this._executingFilePath === undefined) return ts.notImplemented();
+            return this._executingFilePath;
+        }
+
+        public getModifiedTime(path: string) {
+            const stats = this._getStats(path);
+            return stats ? stats.mtime : undefined;
+        }
+
+        public createHash(data: string): string {
+            return data;
+        }
+
+        public realpath(path: string) {
+            try {
+                return this.vfs.realpathSync(path);
+            }
+            catch {
+                return path;
+            }
+        }
+
+        public getEnvironmentVariable(name: string): string | undefined {
+            return this._env && this._env[name];
+        }
+
+        private _getStats(path: string) {
+            try {
+                return this.vfs.statSync(path);
+            }
+            catch {
+                return undefined;
+            }
+        }
+    }
+
+    /**
+     * A fake `ts.ParseConfigHost` that leverages a virtual file system.
+     */
+    export class ParseConfigHost implements ts.ParseConfigHost {
+        public readonly sys: System;
+
+        constructor(sys: System | vfs.FileSystem) {
+            if (sys instanceof vfs.FileSystem) sys = new System(sys);
+            this.sys = sys;
+        }
+
+        public get vfs() {
+            return this.sys.vfs;
+        }
+
+        public get useCaseSensitiveFileNames() {
+            return this.sys.useCaseSensitiveFileNames;
+        }
+
+        public fileExists(fileName: string): boolean {
+            return this.sys.fileExists(fileName);
+        }
+
+        public directoryExists(directoryName: string): boolean {
+            return this.sys.directoryExists(directoryName);
+        }
+
+        public readFile(path: string): string | undefined {
+            return this.sys.readFile(path);
+        }
+
+        public readDirectory(path: string, extensions: string[], excludes: string[], includes: string[], depth: number): string[] {
+            return this.sys.readDirectory(path, extensions, excludes, includes, depth);
+        }
+    }
+
+    /**
+     * A fake `ts.CompilerHost` that leverages a virtual file system.
+     */
+    export class CompilerHost implements ts.CompilerHost {
+        public readonly sys: System;

```

Comment:
Also this one.. Seems like this should just use createCompilerHost API on system instead of having another abstraction.

Reply:
Honestly, it'd be nice if `fakes.CompilerHost` could inherit from `fakes.System` except they differ in their definitions of `useCaseSensitiveFileNames`.